src/util/normalize-manifest/validate.js

```diff
@@ -111,17 +111,6 @@ export function cleanDependencies(info: Object, isRoot: boolean, reporter: Repor
     depTypes.push([type, deps]);
   }
 
-  // check root dependencies for builtin module names
-  if (isRoot) {
-    for (const [type, deps] of depTypes) {
-      for (const name in deps) {
-        if (isBuiltinModule(name)) {
-          warn(reporter.lang('manifestDependencyBuiltin', name, type));
-        }
-      }
-    }
-  }
-
   // ensure that dependencies don't have ones that can collide
   for (const [type, deps] of depTypes) {
```

Comment:
It's a check for whether you're naming your package after a built-in. Arguably if you're doing that you know what you're doing but it affects only a tiny, O(1) number of people so I figured I'd leave that code alone.